### PR TITLE
Perform database migration for execution state on artifact result

### PR DESCRIPTION
**Pull request by kenxu95**
## Describe your changes and why you are making these changes
These are the necessary migrations needed to merge https://github.com/aqueducthq/aqueduct/pull/253

## Related issue number (if any)
ENG-1265

## Checklist before requesting a review
- [x] I have created a descriptive PR title. The PR title should complete the sentence "This PR...".
- [x] I have performed a self-review of my code.
- [ ] I have included a small demo of the changes. For the UI, this would be a screenshot or a Loom video.
- [ ] If this is a new feature, I have added unit tests and integration tests.
- [x] I have run the integration tests locally and they are passing.
- [x] I have run the linter script locally (See `python3 scripts/run_linters.py -h` for usage).
- [ ] All features on the UI continue to work correctly.
- [ ] Added one of the following CI labels:
    - `run_integration_test`: Runs integration tests
    - `skip_integration_test`: Skips integration tests (Should be used when changes are ONLY documentation/UI)

### CODE REVIEW

File: src/golang/cmd/migrator/versions/000014_artifact_result_exec_state_column_backfill/main.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+			)
+			if err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+func Down(ctx context.Context, db database.Database) error {
```
Comment (kenxu95) [inline src/golang/cmd/migrator/versions/000014_artifact_result_exec_state_column_backfill/main.go:30]: Making note to revisit this.

**Review by likawind — COMMENTED** [review]
Everything looks good! I assume you would make sure all lint and tests passes before merging.

One note is to update https://www.notion.so/aqueducthq/Database-Schema-Change-Tracker-c861ce6ba45b43ad87fe7d91eee49df3 and make sure the version number aligns with what @saurav-c have. I don't know if @saurav-c need more time, and which of your versions should merge first.

**Comment by kenxu95** [comment]
@likawind yep I'll sync with @saurav-c tmr. Hopefully we can get this all in then.

**Comment by saurav-c** [comment]
@kenxu95 I've merged by schema change (13), so you can build off of that.

**Review by likawind — APPROVED** [review]

**Pull request merged**